drivers/client_access_control/app/controllers/client_access_control/history_controller.rb

```diff
@@ -234,9 +240,29 @@ def client_needing_processing?
       @client.service_history_enrollments.visible_in_window_to(current_user)
     end
 
-    # Because the PDF is generated by the system user, we need to only include dates that are "visible in the window"
-    private def pdf_enrollment_scope
-      @client.service_history_enrollments.joins(:data_source).merge(::GrdaWarehouse::DataSource.where(visible_in_window: true))
+    private def pdf_enrollment_scope(user = nil)
+      scope = @client.service_history_enrollments
+
+      case ::GrdaWarehouse::Config.get(:verified_homeless_history_method).to_sym
+      when :all_enrollments
+        scope
+      when :visible_in_window
+        scope.joins(:data_source).merge(::GrdaWarehouse::DataSource.where(visible_in_window: true))
+      when :visible_to_user
+        raise 'User is missing' unless user.present?
+
+        scope.visible_in_window_to(user)
+      when :release
+        raise 'User is missing' unless user.present?
+
+        if @client.release_valid?(user.coc_codes)
+          scope
```

Comment:
I wasn't clear on whether this `scope` should be further limited by coc_code. Right now this is saying, if the client has a valid release in _my_ CoC, I include _all_ enrollments regardless of which CoC the enrollment belongs to.

Reply:
That is correct!